src/containerapp/azext_containerapp/custom.py

```diff
@@ -749,7 +749,10 @@ def create_managed_environment(cmd,
                                logs_dynamic_json_columns=False,
                                system_assigned=False,
                                user_assigned=None,
-                               public_network_access=None):
+                               public_network_access=None,
+                               workload_profile_type=None,
+                               workload_profile_name=None,
+                               is_env_for_azml_app=None):
```

Comment:
@harryli0108 
You cannot add these internal params to `create_managed_environment` directly.
Because this function is for command `az containerapp env create`, the added params will be exposed to customer:
You can see them with command `az containerapp env create -h`

![image](https://github.com/user-attachments/assets/bb3cdb36-78b4-4be6-86e2-39002b057c92)
![image](https://github.com/user-attachments/assets/95159c68-86d0-4cb0-aeb7-78d0a37c8b64)

Reply:
You can refer `update_containerapp_logic` to create an internal function for them to avoid the params expose

Reply:
is there a better way to create an env but with consumption gpu wps added into the payload?

Reply:
synced on teams. added `create_managed_env_logic` method to allow up_logic to invoke without exposing this in out-facing methods